### ISSUE TITLE
e2e failing on gke: "Deployment deployment should create new pods"

### ISSUE THREAD

**Issue by ihmccreery** [kubernetes/kubernetes#14752]
Right now, `gke-ci` is failing:

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/deployment.go:32
Expected error:
    <*errors.StatusError | 0xc208dffa00>: {
        ErrStatus: {
            TypeMeta: {Kind: "", APIVersion: ""},
            ListMeta: {SelfLink: "", ResourceVersion: ""},
            Status: "Failure",
            Message: "the server could not find the requested resource",
            Reason: "NotFound",
            Details: {Name: "", Kind: "", Causes: nil, RetryAfterSeconds: 0},
            Code: 404,
        },
    }
    the server could not find the requested resource
not to have occurred
```

This test should not run on GKE, since it requires the experimental API, which is not turned on in GKE.

@caesarxuchao is looking into a way to build test logic for checking if an API is turned on, and doing something, (skipping? failing?,) if it's not.

**Comment by jlowdermilk** [comment]
Fix is #14750

**Comment by caesarxuchao** [comment]
In #14592, I added a new [ServerAPIVersions function](https://github.com/kubernetes/kubernetes/pull/14592/files#diff-f013b1aafa99ed0c479d939400d636e9R433), which returns all the server-supported group/version. It should be easy to use this function to tell if experimental is enabled at the server.
